### PR TITLE
Fix/xpub address validation

### LINKED ISSUE #10940: Importing ETH accounts without checksum validation
There seems to be a bug that allows importing Ethereum addresses into the Trezor Suite mobile app without passing the checksum. Consequently, you can send funds to this address from exchanges that don't perform checksums. This issue can lead to lost funds, especially when users import addresses from images using OCR on their phones.

Image

invalid address example: **0x22a5d8253fACffe4f5eA111cb11cCE1A2887b1b9**

### PR DESCRIPTION
**Pull request by PeKne**
Address validity is checked on account import for address based networks (ETH, SOL, XRP).

## Description
- 00b2e3d41d476bf15d71e6e2a5711280cb3fa495 - utility that checks if the account is address based (does not use XPUB) added
- feb59796c6d4283190261819c4140316dac6e34d 
  - the utility function was used in few existing components to unify the logic to single place of truth
  - translation of these components added
- 4d60a37c3b7276e1132fbe071c6af1c7bd3e0af0 - (**THIS SOLVES THE ISSUE**) the XPUB/address input checks address validity 
 ## Related Issue

Resolve #10940 

## Screenshots:

https://github.com/trezor/trezor-suite/assets/26143964/f72b667a-cb12-4c93-98b0-21e71db96c30

### CODE REVIEW

File: suite-common/wallet-utils/src/accountUtils.ts [hunk 1]
```diff
@@ -962,3 +963,18 @@ export const readUtxoOutpoint = (outpoint: string) => {
 
 export const isSameUtxo = (a: AccountUtxo, b: AccountUtxo) =>
     a.txid === b.txid && a.vout === b.vout;
+
+/**
+ * Returns true if network uses receive address instead of XPUB.
+ */
+export const isAddressBasedNetwork = (networkType: NetworkType) => {
```
Comment (Nodonisko) [inline suite-common/wallet-utils/src/accountUtils.ts:970]: Woudn't be better to have this in as bool in networkConfig?

Comment (PeKne) [reply suite-common/wallet-utils/src/accountUtils.ts:970]: Why not, do you think so?

File: suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx [hunk 2]
```diff
@@ -45,24 +39,38 @@ const cameraStyle = prepareNativeStyle(utils => ({
...
+    const { networkSymbol } = route.params;
+    const { networkType } = networks[networkSymbol];
+
+    const xpubFormValidationSchema = yup.object({
+        xpubAddress: yup
+            .string()
+            .required()
+            .test(
+                'is-valid-address',
+                'Address is not invalid',
```
Comment (juriczech) [inline suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx]: typo

File: suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx [hunk 3]
```diff
@@ -45,24 +39,38 @@ const cameraStyle = prepareNativeStyle(utils => ({
...
 }: StackProps<AccountsImportStackParamList, AccountsImportStackRoutes.XpubScan>) => {
+    const { translate } = useTranslate();
     const { applyStyle } = useNativeStyles();
     const [_, setIsCameraRequested] = useState<boolean>(false);
     const { showAlert, hideAlert } = useAlert();
+
+    const { networkSymbol } = route.params;
+    const { networkType } = networks[networkSymbol];
+
+    const xpubFormValidationSchema = yup.object({
```
Comment (juriczech) [inline suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx]: Not great to have it inside the component. You can use yup context and pass the `networkSymbol` to the schema even when outside of the component.

File: suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx (hunk 2 above)
Comment (PeKne) [reply suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx]: fixed in 8ec74f85a0

File: suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx (hunk 3 above)
Comment (PeKne) [reply suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx]: fixed in 8ec74f85a0

**Review by juriczech — APPROVED** [review]

**Pull request merged**